From a79d39d68ac242c46215fc1be10c16484db408ed Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Wed, 16 Sep 2020 16:47:36 -0700
Subject: [PATCH 1598/1851] misc: xilinx-ai-engine: fix PTR_ERR warning

commit cc7ca6067ebbaea110cbcdc349ef56f1c4e16992 from
https://github.com/Xilinx/linux-xlnx.git

rewrite the eemi ops checking code to fix PTR_ERR warning:

drivers/misc/xilinx-ai-engine/ai-engine-dev.c:402
xilinx_ai_engine_probe() warn: passing a valid pointer to 'PTR_ERR'

Reported-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Wendy Liang <wendy.liang@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/misc/xilinx-ai-engine/ai-engine-dev.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/xilinx-ai-engine/ai-engine-dev.c b/drivers/misc/xilinx-ai-engine/ai-engine-dev.c
index 6afd0c6aecd0..f6ffe61d8377 100644
--- a/drivers/misc/xilinx-ai-engine/ai-engine-dev.c
+++ b/drivers/misc/xilinx-ai-engine/ai-engine-dev.c
@@ -396,11 +396,15 @@ static int xilinx_ai_engine_probe(struct platform_device *pdev)
 	adev->pm_node_id = pm_reg[1];
 
 	adev->eemi_ops = zynqmp_pm_get_eemi_ops();
-	if (IS_ERR(adev->eemi_ops) || !adev->eemi_ops->reset_assert ||
-	    !adev->eemi_ops->get_chipid || !adev->eemi_ops->ioctl) {
+	if (IS_ERR(adev->eemi_ops)) {
 		dev_err(&adev->dev, "failed to get eemi ops.\n");
 		return PTR_ERR(adev->eemi_ops);
 	}
+	if (!adev->eemi_ops->reset_assert || !adev->eemi_ops->get_chipid ||
+	    !adev->eemi_ops->ioctl) {
+		dev_err(&adev->dev, "required eemi ops not found.\n");
+		return -EINVAL;
+	}
 
 	ret = adev->eemi_ops->get_chipid(&idcode, &version);
 	if (ret < 0) {
-- 
2.31.1

